platform/import-export/src/actions/ImportAsJSONAction.js

```diff
@@ -0,0 +1,196 @@
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(['zepto'], function ($) {
+
+    var IMPORT_FORM = {
+        name: "Import as JSON",
+        sections: [
+            {
+                name: "Import A File",
+                rows: [
+                    {
+                        name: 'Select File',
+                        key: 'selectFile',
+                        control: 'importJSONbutton',
+                        required: true,
+                        text: 'Select File'
+                    }
+                ]
+            }
+        ]
+    };
+
+    /**
+     * The ImportAsJSONAction is available from context menus and allows a user
+     * to import a previously exported domain object into any domain object
+     * that has the composition capability.
+     *
+     * @implements {Action}
+     * @constructor
+     * @memberof platform/import-export
+     */
+    function ImportAsJSONAction(exportService, identifierService, dialogService, openmct, context) {
+
+        this.openmct = openmct;
+        this.context = context;
+        this.exportService = exportService;
+        this.dialogService = dialogService;
+        this.identifierService = identifierService;
+        this.instantiate = openmct.$injector.get("instantiate");
+    }
+
+    ImportAsJSONAction.prototype.perform = function () {
+        var importedTree;
+        this.dialogService.getUserInput(IMPORT_FORM, {})
+            .then(function (state) {
+                this.resetButton(IMPORT_FORM);
+                if (this.validateJSON(state.selectFile.body)) {
+                    importedTree = JSON.parse(state.selectFile.body);
+                    this.beginImport(importedTree.openmct);
+                } else {
+                    this.displayError();
+                }
+            }.bind(this), function () {
+                this.resetButton(IMPORT_FORM);
+            }.bind(this));
+    };
+
+    ImportAsJSONAction.prototype.beginImport = function (file) {
+        var parent = this.context.domainObject;
+
+        // Generate tree with newly created ids
+        var tree = this.generateNewTree(file);
```

Comment:
Almost good function naming-- duplicates the comment.  But what would be even better is calling this "generateNewIdentifiers", and getting rid of the comment.